internal/pkg/levee/propagation/propagation.go

```diff
@@ -261,6 +241,53 @@ func (prop *Propagation) visitOperands(n ssa.Node, maxInstrReached map[*ssa.Basi
 	}
 }
 
+func (prop *Propagation) visitCall(call *ssa.Call, maxInstrReached map[*ssa.BasicBlock]int, lastBlockVisited *ssa.BasicBlock) {
+	// Some builtins require special handling
+	if builtin, ok := call.Call.Value.(*ssa.Builtin); ok {
+		switch builtin.Name() {
+		// The values being appended cannot be tainted.
+		case "append":
+			// The slice argument needs to be tainted because if its underlying array has
+			// enough remaining capacity, the appended values will be written to it.
+			prop.visitCallArg(call.Call.Args[0], maxInstrReached, lastBlockVisited)
+			// The returned slice is tainted if either the slice argument or the values
+			// are tainted, so we need to visit the referrers.
+			prop.visitReferrers(call, maxInstrReached, lastBlockVisited)
+			return
+
+		// Only the first argument (dst) can be tainted. (The src cannot be tainted.)
+		case "copy":
+			prop.visitCallArg(call.Call.Args[0], maxInstrReached, lastBlockVisited)
+			return
+
+		// The builtin delete(m map[Type]Type1, key Type) func does not propagate a taint.
+		case "delete":
+			return
+		}
+	}
```

Comment:
I think we should break this out into its own function like visitCallToBuiltin or visitBuiltinCall or visitBuiltin.

Reply:
Agreed. Done.